### ISSUE TITLE
Fit the spline doesn't work correctly. It will cause *Signal: SIGABRT (Aborted)* issue.

### ISSUE THREAD

**Issue by Lurvelly** [brytsknguyen/slict#24]
Hi, Thien-Minh Nguyen

Thanks for sharing your great work! I have learned a lot.

I found there may cause *Signal: SIGABRT (Aborted)* issue, if I turn the parameter "fit_spline" to 1. 
![screenshot-20240930-111113](https://github.com/user-attachments/assets/82e6b134-0dde-436d-99cc-79847c3bb898)

I have debugged this issue and found there is a fault usage of std::thread in your code below:
https://github.com/brytsknguyen/slict/blob/a733e7fc48c77bda6a4d7efd944c563185fa9b18/src/Estimator.cpp#L1403-L1412

You have allocated a thread object on the stack and the thread will be destroyed before a new loop.

So I have to trun the parameter "fit_spline" to 0, so that we can avoid create the issue.
```yaml
fit_spline:         0
```

Here is a simple code to reproduct the *Signal: SIGABRT (Aborted)* issue:
```c++
//
// Created by Liuyang Li on 30/09/24.
//
#include <iostream>
#include <thread>

void test()
{
    for(int i = 0; i < 10000000; i++)
    {
        std::cout << i << std::endl;
    }
}

int main()
{
    std::thread t1(&test);
}
```

Sorry for my poor understand your SLAM system, I'm afraid of influencing the SLAM progress, so I can't fix this issue by myself. 
Maybe you can use "join()" at the end of the loop 

![Screenshot from 2024-09-30 11-39-56](https://github.com/user-attachments/assets/f718a3d1-5c0c-417b-aeff-99f0ee6920f7)

or just use "detach()" after creating the thread.
```c++
            // Fit the spline at the ending segment to avoid high cost
            std::thread threadFitSpline;
            if (fit_spline)
            {
                static bool fit_spline_enabled = false;
                if (SwTimeStep.size() >= WINDOW_SIZE && !fit_spline_enabled)
                    fit_spline_enabled = true;
                else if(fit_spline_enabled)
                {
                    threadFitSpline = std::thread(&Estimator::FitSpline, this);
                    threadFitSpline.detach();
                }
            }
```
